internal/kafka/internal/migrations/20230109140000_update_expires_at_zero_value.go

```diff
@@ -0,0 +1,34 @@
+package migrations
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+func updateExpiresAtZeroValueFromKafkaRequests() *gormigrate.Migration {
+	nullTime := sql.NullTime{Time: time.Time{}, Valid: false}
+	zeroTime := time.Time{}
+
+	return &gormigrate.Migration{
+		ID: "20230109140000",
+		Migrate: func(tx *gorm.DB) error {
+			err := tx.Table("kafka_requests").Where("expires_at = ?", zeroTime).Where("deleted_at IS NULL").Update("expires_at", nullTime).Error
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+		Rollback: func(tx *gorm.DB) error {
+			err := tx.Table("kafka_requests").Where("expires_at = ?", nullTime).Where("deleted_at IS NULL").Update("expires_at", zeroTime).Error
```

Comment:
I believe this is not an _actual_ rollback since it will update **all** the rows where `expires_at` is `null` to the `zeroTime` value (even the rows where `expires_at` was already null before this migration started.

However, it should only be an issue if I'm missing something.

Reply:
Before the migration no columns with the NULL value should exist. If the migration fails then those columns would be reverted. There might be some edge case for an instance being created but in the middle but there is not much we can do. The alternative would be not doing any rollback.